### PR TITLE
Add some logs about concurrency

### LINKED ISSUE #8215: Add some logs about concurrency

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #8215

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by windtalker — APPROVED** [review]
LGTM

**Review by gengliqi — APPROVED** [review]

File: dbms/src/Flash/Statistics/JoinImpl.cpp [hunk 1]
```diff
@@ -20,16 +20,18 @@ namespace DB
...
-        R"("join_build_inbound_rows":{},"join_build_inbound_blocks":{},"join_build_inbound_bytes":{},"join_build_inbound_allocated_bytes":{}, "join_build_execution_time_ns":{})",
+        R"("peak_build_bytes_usage":{},"build_side_child":"{}","is_spill_enabled":"{}","is_spilled":"{}",)"
+        R"("join_build_inbound_rows":{},"join_build_inbound_blocks":{},"join_build_inbound_bytes":{},)"
+        R"("join_build_inbound_allocated_bytes":{},"join_build_concurrency":{},"join_build_execution_time_ns":{})",
         peak_build_bytes_usage,
         build_side_child,
-        is_spill_enabled,
-        is_spilled,
+        is_spill_enabled ? "true" : "false",
+        is_spilled ? "true" : "false",
```
Comment (JaySon-Huang) [inline dbms/src/Flash/Statistics/JoinImpl.cpp:29]: why we need this? won't the fmtlib turn it into "true"/"false" for boolean values?

File: dbms/src/Flash/Statistics/JoinImpl.cpp [hunk 2]
```diff
@@ -20,16 +20,18 @@ namespace DB
 void JoinStatistics::appendExtraJson(FmtBuffer & fmt_buffer) const
 {
     fmt_buffer.fmtAppend(
-        R"("peak_build_bytes_usage":{},"build_side_child":"{}","is_spill_enabled":{},"is_spilled":{})"
-        R"("join_build_inbound_rows":{},"join_build_inbound_blocks":{},"join_build_inbound_bytes":{},"join_build_inbound_allocated_bytes":{}, "join_build_execution_time_ns":{})",
+        R"("peak_build_bytes_usage":{},"build_side_child":"{}","is_spill_enabled":"{}","is_spilled":"{}",)"
```
Comment (SeaRise) [inline dbms/src/Flash/Statistics/JoinImpl.cpp:23]:
```suggestion
        R"("peak_build_bytes_usage":{},"build_side_child":"{}","is_spill_enabled":{},"is_spilled":{},)"
```

File: dbms/src/Flash/Statistics/JoinImpl.cpp (hunk 1 above)
Comment (SeaRise) [inline dbms/src/Flash/Statistics/JoinImpl.cpp:29]:
```suggestion
        is_spill_enabled,
        is_spilled,
```

Comment (SeaRise) [reply dbms/src/Flash/Statistics/JoinImpl.cpp]: It's really unnecessary

**Pull request merged**